src/camera_utils.py

```diff
@@ -30,7 +30,17 @@ def project_world_point_to_image(camera: Camera, point: np.ndarray) -> np.ndarra
     Returns:
         np.ndarray: [u, v] pixel coordinates corresponding to the point.
     """
-    raise NotImplementedError()
+    scene_point_X = float(point[0])
+    scene_point_Y = float(point[1])
+    scene_point_Z = float(point[2])
+
+    image_point_x = float(camera.fx * (scene_point_X / scene_point_Z))
+    image_point_y = float(camera.fy * (scene_point_Y / scene_point_Z))
+
+    pixel_u = float(image_point_x + camera.cx)
+    pixel_v = float(image_point_y + camera.cy)
```

Comment:
No need to type-cast everything to float. Values are float so operations on them will result in floats.